Tests/Linq/Generators/ExpressionMethodGeneratorTests.cs

```
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using LinqToDB;
+using LinqToDB.Expressions;
+using LinqToDB.Mapping;
+using NUnit.Framework;
+
+namespace Tests.Generators
+{
+	[TestFixture]
+	public partial class ExpressionMethodGeneratorTests : TestBase
+	{
+		public class PersonDto
+		{
+			public int Id { get; set; }
+			public string Name { get; set; } = null!;
+		}
+
+		public partial class ExprPerson
+		{
+			public int Id { get; set; }
+			public string Name { get; set; } = null!;
+
+			[GenerateExpressionMethod(MethodName = "Test1")]
+			public PersonDto PersonDto1 =>
+				new PersonDto { Id = Id, Name = Name, };
+
+			[GenerateExpressionMethod(MethodName = "Test2")]
+			public PersonDto PersonDto2
+			{
+				get => new PersonDto { Id = this.Id, Name = this.Name, };
+			}
+
+			[GenerateExpressionMethod(MethodName = "Test3")]
+			public PersonDto PersonDto3
+			{
+				get
+				{
+					return new PersonDto { Id = Id, Name = Name, };
+				}
+			}
+
+			public static ExprPerson[] GenerateTestData() =>
+				Enumerable.Range(0, 20)
+					.Select(i => new ExprPerson { Id = i, Name = $"Person{i}", })
+					.ToArray();
+		}
+
+		[GenerateExpressionMethod]
+		public static PersonDto ToDtoReturn(ExprPerson person)
+		{
+			return new PersonDto
+			{
+				Id = person.Id,
+				Name = person.Name,
+			};
+		}
+
+		[GenerateExpressionMethod]
+		public static PersonDto ToDtoArrow(ExprPerson person)
+			=> new PersonDto
+			{
+				Id = person.Id,
+				Name = person.Name
+			};
```

There are also needed tests for overloads.

---

Good point - overloads create clashing expression method functions unless you use `MethodName`. What would you prefer as proper behavior here? a) let the compile fail, or b) generate an L2DB error, or c) something else? (I vote for b) myself.)

---

Generate method name using signature. MethodName + Full Parameter type names is the best approach I think.

---

Done!